querydsl-lucene5/src/test/java/com/querydsl/lucene5/LuceneQueryTest.java

```diff
@@ -0,0 +1,809 @@
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.DoubleDocValuesField;
+import org.apache.lucene.document.DoubleField;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.Field.Store;
+import org.apache.lucene.document.IntField;
+import org.apache.lucene.document.NumericDocValuesField;
+import org.apache.lucene.document.SortedDocValuesField;
+import org.apache.lucene.document.TextField;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.sandbox.queries.DuplicateFilter;
+import org.apache.lucene.search.Filter;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Sort;
+import org.apache.lucene.store.RAMDirectory;
+import org.apache.lucene.util.BytesRef;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import com.google.common.collect.Sets;
+import com.querydsl.core.NonUniqueResultException;
+import com.querydsl.core.QueryException;
+import com.querydsl.core.QueryModifiers;
+import com.querydsl.core.QueryResults;
+import com.querydsl.core.types.ParamNotSetException;
+import com.querydsl.core.types.dsl.Expressions;
+import com.querydsl.core.types.dsl.NumberPath;
+import com.querydsl.core.types.dsl.Param;
+import com.querydsl.core.types.dsl.StringPath;
+
+/**
+ * Tests for LuceneQuery
+ *
+ * @author vema
+ *
+ */
+public class LuceneQueryTest {
+
+    private LuceneQuery query;
+    private StringPath title;
+    private NumberPath<Integer> year;
+    private NumberPath<Double> gross;
+
+    private final StringPath sort = Expressions.stringPath("sort");
+
+    private RAMDirectory idx;
+    private IndexWriter writer;
+    private IndexSearcher searcher;
+    private Document doc = new Document();
+    private Field titleField = null;
+    private SortedDocValuesField titleSortedField;
+    private TextField authorField = null;
+    private SortedDocValuesField authorSortedField;
+    private TextField textField = null;
+    private IntField yearField = null;
+    private DoubleField grossField = null;
+    private SortedDocValuesField textSortedField;
+    private NumericDocValuesField yearSortedField;
+    private DoubleDocValuesField grossSortedField;
+
+    private Document createDocument(final String docTitle,
+            final String docAuthor, final String docText, final int docYear,
+            final double docGross) {
+        // Reusing field for performance
```

Comment:
I can change the method name if you want to. Reusing the field already created is recommended when performance is needed. 
We probably don't need performance, but it isn't an issue. They all will be different document in the index.